src/test/ui/rfc-2008-non-exhaustive/struct.stderr

```diff
@@ -2,7 +2,7 @@ error[E0423]: cannot initialize a tuple struct which contains private fields
   --> $DIR/struct.rs:20:14
    |
 LL |     let ts = TupleStruct(640, 480);
-   |              ^^^^^^^^^^^ constructor is not visible here due to private fields
```

Comment:
The reason we actually had this span_label before was because this was one of the cases handled by the `DefKind::Ctor` arm I removed above which you can see just always passed in an empty vec for the field visibilities. Now that we pass in the correct value there, not printing this message is actually correct since none of the fields are actually private! The reason for the error is because of `#[non_exhaustive]` which unfortunately isn't handled very well right now. I can hopefully get to improving the non-exhaustive errors and diagnostics.

Reply:
In that case we should _also_ change the error message.

I would like us to keep the `constructor is not visible here due to private fields` label regardless. If you look at the other output it is missing as well. Don't get me wrong, the new note is fantastic, but missing primary labels are a potential problem: some people have trained themselves to _not_ read the main message, in other cases tools that display tooltips in users' code rely on the primary label instead of the message because it is usually better in context. I made a mockup on the other test output of what I'd like to see in the future. I'd be inclined to merge this PR if you don't intend on continuing work, but I'd like a ticket open for the follow up tweaks.